include/tscore/Regex.h

```diff
@@ -45,16 +47,46 @@ class Regex
 {
 public:
   /// Default number of capture groups.
-  static constexpr size_t DEFAULT_GROUP_COUNT = 30;
+  static constexpr size_t DEFAULT_GROUP_COUNT = 10;
 
   Regex() = default;
+  ~Regex();
+
+  /** Compile the @a pattern into a regular expression.
+   *
+   * @param pattern Source pattern for regular expression (null terminated).
+   * @param flags Compilation flags.
+   * @return @a true if compiled successfully, @a false otherwise.
+   *
+   * @a flags should be the bitwise @c or of @c REFlags values.
+   */
   bool compile(const char *pattern, const unsigned flags = 0);
-  // It is safe to call exec() concurrently on the same object instance
-  bool exec(const char *str);
-  bool exec(const char *str, int length);
-  bool exec(const char *str, int length, int *ovector, int ovecsize);
+
+  /** Execute the regular expression.
+   *
+   * @param str String to match against.
+   * @return @c true if the patter matched, @a false if not.
+   *
+   * It is safe to call this method concurrently on the same instance of @a this.
+   */
+  bool exec(std::string_view const &str);
```

Comment:
Why pass by const reference rather than by value?  https://godbolt.org/z/IEut0X

Reply:
Why not? Walt, we've discussed this over and over, I'm not sure how to be clearer. I use constant reference if it is to be passed to another function, thereby avoiding in intermediate functions
```
        mov     qword ptr [rsp + 8], rdi
        mov     qword ptr [rsp + 16], rsi
```
The problem with your examples is that in such simple situations there is far more scope for optimization, e.g. because registers are not being used by other computations they are available as dedicated stashes for values. And even then, the performance difference (if any) is so tiny as to be less than negligible (a few hundred picoseconds at most) and even then it depends on tiny details in the code.

Reply:
Sorry I don't remember discussing this before.  If you need two registers to pass the actual string_view instead of one to pass an address, that may mean an extra push to the stack to preserve a register value.  But I'm not sure about that.  I think usually ABIs assume that registers used to pass parameters might be used as scratch space (without first saving) by called functions (even when they are not used as parameters).  When you pass anything by address (non-inlined) you guarantee (barring LTO/WPO) that it can't be a register-only object.  But the performance issue is of dubious value.  I think of string_view as sort of a smart pointer, an improvement to const char *.  The extra level of indirection seems icky somehow.

Reply:
Well, the C++ standards committee view seems that `string_view` is more of a container than a pointer / iterator (one of the design philosophy differences between that and `TextView`). As noted, though, I use the reference when I'm not going to touch the contents, but just pass it on, so it doesn't matter if it's register only or not.